CoreWordCloud/src/au/gov/asd/tac/constellation/views/wordcloud/ui/WordCloudController.java

```diff
@@ -0,0 +1,426 @@
+        controllerIsInitialising = true;
+
+        // Fire a new active graph event the first time a pane is attached
+        parent.handleNewGraph(GraphManager.getDefault().getActiveGraph());
+        controllerIsInitialising = false;
+    }
+
+    /**
+     * Called by the WordCloudPane being controller when a word is clicked.
+     * Modifies the controller WordCloud appropriately and then select the
+     * relevant elements on the graph
+     */
+    public void alterSelection(final String word, final boolean accumulativeSelection, final boolean deselect) {
+        // Alter the selection appropriately 
+        if (!accumulativeSelection) {
+            cloud.singleWordSelection(word);
+        } else if (deselect) {
+            cloud.removeWordFromSelection(word);
+        } else {
+            cloud.addWordToSelection(word);
+        }
+
+        updateSelectedWordsOnPane();
+        selectElements();
+    }
+
+    public void selectElements() {
+        // Retrieve the elements to be selected from the WordCloud 
+        final Set<Integer> elementsToSelect = cloud.getElementsCorrespondingToSelection();
+        final GraphElementType graphElementType = cloud.getElementType();
+
+        // Run a simple edit plugin to select the elements on the current graph 
+        PluginExecution.withPlugin(new SimpleEditPlugin("Word Cloud: Select Elements") {
+            @Override
+            public void edit(final GraphWriteMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException {
+                final int selectedAttr = graph.getAttribute(graphElementType, SELECTED_ATTRIBUTE);
+                final int vertSelectedAttr = graph.getAttribute(GraphElementType.VERTEX, SELECTED_ATTRIBUTE);
+                final int transSelectedAttr = graph.getAttribute(GraphElementType.TRANSACTION, SELECTED_ATTRIBUTE);
+
+                if (selectedAttr == Graph.NOT_FOUND) {
+                    return;
+                }
+
+                if (vertSelectedAttr != Graph.NOT_FOUND) {
+                    for (int i = 0; i < graph.getVertexCount(); i++) {
+                        graph.setBooleanValue(vertSelectedAttr, graph.getVertex(i), false);
+                    }
+                }
+                if (transSelectedAttr != Graph.NOT_FOUND) {
+                    for (int i = 0; i < graph.getTransactionCount(); i++) {
+                        graph.setBooleanValue(transSelectedAttr, graph.getTransaction(i), false);
+                    }
+                }
+
+                elementsToSelect.forEach(el -> 
+                    graph.setBooleanValue(selectedAttr, el, true));
+            }
+        }).executeLater(graph);
+    }
+
+    /**
+     * Manages a new graph becoming active in the application
+     */
+    public void updateActiveGraph(final Graph graph) {
+        vertTextAttributes = new ArrayList<>(EMPTY_STRING_LIST);
+        transTextAttributes = new ArrayList<>(EMPTY_STRING_LIST);
```

Comment:
see previous comment for first initialisation. Not sure whether its better to simply clear the existing lists and re-add empty string rather than creating a new instance